src/components/Graph/Graph.tsx

```diff
@@ -92,13 +98,31 @@ const Graph: FC<GraphProps> = memo(({ pathway, interactive = true, expandCurrent
     }, {});
   }, [layoutKeys]);
 
-  const [expanded, _setExpanded] = useState<{ [key: string]: boolean | undefined }>(
-    initialExpandedState
-  );
+  const [expanded, _setExpanded] = useState({
+    expandedArray: initialExpandedState,
+    lastSelectedNode: ''
+  });
 
   const setExpanded = useCallback((key: string, expand?: boolean): void => {
     _setExpanded(prevState => {
-      return { ...prevState, [key]: !prevState[key] };
+      if (prevState.expandedArray[key] === undefined) {
+        // selecting a new node
+        const newExpandedArray = {
+          ...prevState.expandedArray,
+          [key]: !prevState.expandedArray[key]
+        };
+        return { ...prevState, expandedArray: newExpandedArray, lastSelectedNode: key };
+      } else if (prevState.lastSelectedNode !== key && prevState.expandedArray[key] === true) {
+        // selected a different but already opened node
+        return { ...prevState, expandedArray: prevState.expandedArray, lastSelectedNode: key };
+      } else {
+        // selected the same node or a different but collapsed node
+        const newExpandedArray = {
+          ...prevState.expandedArray,
+          [key]: !prevState.expandedArray[key]
+        };
+        return { ...prevState, expandedArray: newExpandedArray, lastSelectedNode: key };
+      }
```

Comment:
It looks like the first and third blocks are exactly the same, and all 3 instances return the same structure, so could this be reduced to:
```suggestion
      let newExpandedArray = prevState.expandedArray;

      // if clicking on a node that was not already expanded,
      // or re-clicking on the current already open node, toggle it
      if (!prevState.expandedArray[key] || prevState.lastSelectedNode === key) {
        newExpandedArray = {
          ...prevState.expandedArray,
          [key]: !prevState.expandedArray[key]
        };
      }

      return { ...prevState, expandedArray: newExpandedArray, lastSelectedNode: key };
```